src/component/page/bruker-detaljer/tildel-veileder-modal/TildelVeilederModal.tsx

```diff
@@ -0,0 +1,144 @@
+import { Alert, Button, Detail } from '@navikt/ds-react'
+import React, { useEffect, useMemo, useState } from 'react'
+import { TilgjengeligVeileder, Veileder } from '../../../../api/data/veileder'
+import { MultiValue, SingleValue } from 'react-select'
+import { lagBrukerNavn } from '../../../../utils/bruker-utils'
+
+import styles from './TildelVeilederModal.module.scss'
+import { isPending, isResolved, usePromise } from '../../../../utils/use-promise'
+import { AxiosResponse } from 'axios'
+import { hentTilgjengeligeVeiledere, tildelVeilederForDeltaker } from '../../../../api/tiltak-api'
+import { TiltakDeltakerDetaljer } from '../../../../api/data/deltaker'
+import { SelectField, SelectOption } from '../../../felles/select-field/SelectField'
+import { BaseModal } from '../../../felles/base-modal/BaseModal'
+
+
+interface Props {
+	deltaker: TiltakDeltakerDetaljer
+	veileder: Veileder | undefined
+	medveiledere: Veileder[]
+	open: boolean
+	onClose: () => void
+	onSubmit: (veiledere: Veileder[]) => void
+}
+
+const maksMedveiledere = 3
+
+export const TildelVeilederModal = (props: Props): React.ReactElement => {
+	const [ tilgjengeligeVeiledere, setTilgjengeligeVeiledere ] = useState<TilgjengeligVeileder[]>([])
+	const [ veilederValg, setVeilederValg ] = useState<SelectOption[]>()
+	const [ veileder, setVeileder ] = useState<TilgjengeligVeileder | undefined>()
+	const [ medveiledere, setMedveiledere ] = useState<TilgjengeligVeileder[]>([])
+	const [ veilederError, setVeilederError ] = useState(false)
+
+	const tildelVeilederePromise = usePromise<void>()
+	const tilgjengeligeVeilederePromise = usePromise<AxiosResponse<TilgjengeligVeileder[]>>(
+		() => hentTilgjengeligeVeiledere(props.deltaker.gjennomforing.id)
+	)
+
+
+	useMemo(() => {
+		if (isResolved(tilgjengeligeVeilederePromise)) {
+			setTilgjengeligeVeiledere(tilgjengeligeVeilederePromise.result.data)
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ tilgjengeligeVeilederePromise.result ])
+
+	useEffect(() => {
```

Comment:
Isteden for dette så kan du initiere staten med defaultverdiene som ligger i propsene. Det er mer lesbar fordi da ser man hva default verdien er med en gang isteden for å tråle igjennom mange blokker med kode i forskjellige useEffects

Reply:
Jeg kan prøve igjen, men jeg tror ikke props.veileder nødvendigvis finnes på første render.

Reply:
Prøvde å skrive om litt, men finner ingen bedre løsning enn å bruke en useEffect for dette, da de ikke blir satt riktig på første render.

Reply:
Ja, du har rett. Grunnen er at komponenten som kaller denne, ikke sjekker om promisen er resolved før denne komponenten rendres. Så det vil kunne skje at man trykker på knappen uten at man har fått tilbake svar fra backend